etc/convert-test-results.swift

```diff
@@ -0,0 +1,332 @@
+import Foundation
+
+/// Represents a test suite.
+struct TestSuite {
+    /// Name of the test suite.
+    let name: String
+
+    /// Total execution time for the test suite, in seconds.
+    let time: TimeInterval
+
+    /// Tests in the suite.
+    let tests: [TestCase]
+
+    /// Count of tests in the suite.
+    let count: Int
+
+    /// Count of failed tests in the suite.
+    let failureCount: Int
+
+    /// Converts this test suite to XML.
+    func toXML() -> String {
+        var output =
+        """
+        <testsuite tests="\(self.count)" failures="\(self.failureCount)" errors="0" time="\(self.time)" name="\(self.name)">\n
+        """
+
+        for test in self.tests {
+            output += test.toXML()
+        }
+
+        output += "</testsuite>\n"
+        return output
+    }
+}
+
+/// Represents a test case.
+struct TestCase {
+    /// The name of the class this test case belongs to.
+    let className: String
+
+    /// The name of this test case.
+    let name: String
+
+    /// The time the test case took to run, in seconds.
+    let time: TimeInterval
+
+    /// Failure message produced by the test case, if any.
+    let failure: String?
+
+    func toXML() -> String {
+        var output =
+            """
+            <testcase classname="\(self.className)" name="\(self.name)" time="\(self.time)">\n
+            """
+
+        if let failure = self.failure {
+            // hack to replace disallowed XML characters with very similar unicode ones.
+            // evergreen doesn't render the escaped XML characters correctly so this preserves
+            // readability while keeping the XML valid.
+            let escapedFailure = failure
+                .replacingOccurrences(of: "\"", with: "＂")
+                .replacingOccurrences(of: "'", with: "＇")
+                .replacingOccurrences(of: "<", with: "﹤")
```

Comment:
TIL there are "small" versions of many unicode symbols, lol. the "right" way to do this in XML is to use e.g. `&quot;` instead of `"` but that doesn't end up being very readable in the HTML logs.